File: app/src/main/java/com/mlsdev/mlsdevstore/data/remote/service/BrowseService.java
package com.mlsdev.mlsdevstore.data.remote.service;

import com.mlsdev.mlsdevstore.data.model.item.Item;
import com.mlsdev.mlsdevstore.data.model.item.SearchResult;

import java.util.Map;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.QueryMap;

public interface BrowseService {

    @GET("/buy/browse/v1/item_summary/search")
    Single<SearchResult> searchItemsByCategoryId(@QueryMap Map<String, String> queries);

    @GET("/buy/browse/v1/item/{id}")
    Single<Item> getItem(@Path("id") String itemId);

}

File: app/src/main/java/com/mlsdev/mlsdevstore/data/remote/service/BrowseService.kt
package com.mlsdev.mlsdevstore.data.remote.service

import com.mlsdev.mlsdevstore.data.model.item.Item
import com.mlsdev.mlsdevstore.data.model.item.SearchResult

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.QueryMap

interface BrowseService {

    @GET("/buy/browse/v1/item_summary/search")
    fun searchItemsByCategoryId(@QueryMap queries: Map<String, String>): Single<SearchResult>

    @GET("/buy/browse/v1/item/{id}?fieldgroups=PRODUCT")
    fun getItem(@Path("id") itemId: String): Single<Item>

}
